### DISCUSSION TITLE
Orquesta Workflow Testing

### DISCUSSION THREAD

**Discussion by jpavlav** [StackStorm/st2#6255] [Q&A]
I'm wondering if there is a way to test orquesta workflows with mocked data. I realize we could stand up mocked services for all of the things that might be talked to in a given workflow, but I'm talking more along the lines of unittests, where we can provide mocked data as the answer for various actions in a workflow and ensure the workflow works as expected.

**Comment by nzlosh** [comment]
https://docs.stackstorm.com/development/pack_testing.html  From what I remember, there's some structures in the Orquesta code to allow for dry runs, but I've no idea what state it's in. To give some perspective, this feature has been dormant since 2019 and little interest has been expressed.  I assume most people test workflows against development environments.

**Reply by jpavlav** [reply]
Ah I see, I found this:

https://github.com/StackStorm/community/issues/27

And was hopeful that maybe there was something lurking around that I was unaware of. We have staging environments but testing our workflows in that way would be tough for _reasons_. Plus, there is the issue of testing with real services can be a pretty slow feedback loop. I'll Leave this open for now, I'll try to see what I can glean from the `orquesta` repo. Maybe I can come up with something.

**Reply by jpavlav** [reply]
@guzzijones Do you happen to have a branch around with this `orquesta-test` binary/interface?
